src/runtime/lib/telemetry/index.ts

```diff
@@ -150,6 +218,35 @@ export class TelemetryClient {
             });
     }
 
+    /**
+     * check data and timestamp for updating telemetry data
+     * @returns 
+     */
+    private shouldUpdateData(): boolean {
+        if (ext.context.globalState.get(TELEMETRY_DATA) == undefined && 
+        (ext.context.globalState.get(TELEMETRY_DATA) as []).length == 0 ) {
```

Comment:
Guess this is not a `&&` but a `||` check? Since once the variable is undefined, there won't be a length check & vice-versa, so technically this if clause will always be `false`.